tests/test_logs.py

```diff
@@ -0,0 +1,372 @@
+    scope_logs.log_records.append(log_record)
+
+    # Create resource logs
+    resource_logs = ResourceLogs()
+    resource_logs.resource.CopyFrom(resource)
+    resource_logs.scope_logs.append(scope_logs)
+
+    # Create export request
+    export_request = ExportLogsServiceRequest()
+    export_request.resource_logs.append(resource_logs)
+
+    return export_request.SerializeToString()
+
+
+@pytest.fixture
+def otlp_logs_json(service_name, environment, version, host_name, log_message, trace_id, span_id):
+    """JSON representation of OTLP logs payload."""
+    return {
+        "resource_logs": [
+            {
+                "resource": {
+                    "attributes": [
+                        {"key": "service.name", "value": {"string_value": service_name}},
+                        {"key": "deployment.environment", "value": {"string_value": environment}},
+                        {"key": "service.version", "value": {"string_value": version}},
+                        {"key": "host.name", "value": {"string_value": host_name}},
+                    ]
+                },
+                "scope_logs": [
+                    {
+                        "log_records": [
+                            {
+                                "body": {"string_value": log_message},
+                                "trace_id": trace_id.hex() if trace_id else "",
+                                "span_id": span_id.hex() if span_id else "",
+                            }
+                        ]
+                    }
+                ],
+            }
+        ]
+    }
+
+
+@pytest.fixture
+async def otlp_http_agent(agent_app, aiohttp_client):
+    """Test client for OTLP HTTP app."""
+    # Get the shared agent instance from the main app
+    agent = agent_app.app["agent"]
+    otlp_http_app = make_otlp_http_app(agent)
+
+    # Create a client for the OTLP HTTP app
+    client = await aiohttp_client(otlp_http_app)
+    yield client
+
+
+# Future: GRPC support fixtures
+# @pytest.fixture
+# async def otlp_grpc_client():
+#     """Test client for OTLP GRPC server."""
+#     # Create GRPC channel and LogsService stub for port 4317
+#     pass
+#
+#
+# @pytest.fixture
+# async def otlp_grpc_agent(agent_app):
+#     """Test client for OTLP GRPC server that forwards to HTTP."""
+#     # Return GRPC client for testing GRPC→HTTP forwarding
+#     pass
+
+
+# Future: GRPC support
+# @pytest.fixture
+# def otlp_grpc_url(testagent_url):
+#     """URL for OTLP GRPC logs endpoint."""
+#     pass
+
+
+@pytest.fixture
+def otlp_http_url(testagent_url):
+    """URL for OTLP HTTP logs endpoint."""
+    parsed_url = urlparse(testagent_url)
+    return f"{parsed_url.scheme}://{parsed_url.hostname}:{OTLP_HTTP_PORT}"
+
+
+@pytest.fixture
+def otlp_http_client(otlp_http_url):
+    """Test Agent client for retrieving logs from the OTLP HTTP endpoint."""
+    return TestAgentClient(otlp_http_url)
+
+
+async def test_logs_endpoint_basic_http(otlp_http_agent, otlp_logs):
```

Comment:
For these test cases, can you parametrize the following inputs and then at the beginning of the test method you could call `otlp_logs` to generate the request data from these inputs?

- service_name
- environment
- version
- log_message (this could even be an array to test sending multiple log messages from the same scope in the same request)

Reply:
Basically, I don't think the inputs should be test fixtures since we could theoretically test a variety of combinations of them

Reply:
Oh we can parameterize fixtures. I provided an example on another test. This allows us to create default values for otlp_logs that can be overriden in other tests.